bioconda_utils/lint_functions.py

```diff
@@ -0,0 +1,186 @@
+import os
+import glob
+
+
+def _subset_df(recipe, meta, df):
+    """
+    Helper function to get the subset of `df` for this recipe.
+    """
+    name = meta['package']['name']
+    version = meta['package']['version']
+
+    # Some recipes have an empty build section, so it'll be None and we can't
+    # do a chained get.
+    build_number = 0
+    build_section = meta.get('build')
+    if build_section:
+        build_number = int(build_section.get('number', 0))
+    return df[
```

Comment:
True, it would be cleaner to do `df.loc[name, version, build_number]`, but only if all these keys exist. In the case where a build number was bumped and so doesn't exist in the df, you'd have to catch the KeyError and then return an empty dataframe. So I think I'd prefer to keep it flat for now, unless I'm missing something in pandas where it handles missing hierarchical keys gracefully.